custom/logistics/tasks.py

```diff
@@ -9,32 +9,27 @@
 from celery.task.base import task
 
 
-@task
-def stock_data_task(domain, endpoint, apis, test_facilities=None):
-    # checkpoint logic
-    start_date = datetime.today()
+def get_or_create_checkpoint(domain):
     try:
         checkpoint = StockDataCheckpoint.objects.get(domain=domain)
-        api = checkpoint.api
-        date = checkpoint.date
-        limit = checkpoint.limit
-        offset = checkpoint.offset
-        location = checkpoint.location
         if not checkpoint.start_date:
-            checkpoint.start_date = start_date
+            checkpoint.start_date = datetime.today()
```

Comment:
checkpoint.start_date means date when pending migration was started. checkpoint.date is basically date of last successful migration (probably it's worth to change name of this field).